### PR TITLE
Client: Update Section Header component to Typescript file

### PR DESCRIPTION
**Pull request by kirahsapong**
Why: To reap the benefits of TypeScript (such as added safety and automation) for this file. Also, a good way to practice creating a PR for Jetpack.

`cardBadge` prop and its use in `dash-item` should be removed in a second PR. This prop was removed in [a previous PR](https://github.com/Automattic/jetpack/pull/9732). 

`SectionHeader` component should make use of semantic `h` level tags in a follow-up PR for improved a11y. 

#### Changes proposed in this Pull Request:

* Convert `SectionHeader` component from Javascript to Typescript file
* Add `cardBadge` optional prop back into component for backwards compatibility

#### Other information:

- [x] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:
1. Pull up Jetpack Settings screen at `/wp-admin/admin.php?page=jetpack#/settings`
2. Section Headings should be visible per the screenshot below:
Screen Shot 2022-10-05 at 12 40 26 PM

### CODE REVIEW

**Comment by jeyip** [comment]
For the typed section header file, I'm seeing a "can't find module" error. Did you / do you happen to see the same thing?

Screen Shot 2022-10-06 at 12 19 14 AM

**Comment by jeyip** [comment]
## Testing

### Requirements
- [x] Section Headings should be visible

### Browsers
- [x] Chrome

File: projects/plugins/jetpack/_inc/client/components/section-header/index.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import classNames from 'classnames';
+import Card from 'components/card';
+import React from 'react';
+
+import './style.scss';
+
+interface SectionHeaderProps {
+	label?: string;
```
Comment (jeyip) [inline projects/plugins/jetpack/_inc/client/components/section-header/index.tsx]: `NB`: Should we allow this to be optional? If so, maybe it should have a default value of an empty string?

**Review by jeyip — COMMENTED** [review]
Changes generally look good 🎉  -- just curious about https://github.com/Automattic/jetpack/pull/26644#issuecomment-1269476162

+1 from me when the comment is resolved

File: projects/plugins/jetpack/_inc/client/components/section-header/index.tsx (hunk 1 above)
Comment (kirahsapong) [reply projects/plugins/jetpack/_inc/client/components/section-header/index.tsx]: @jeyip Good catch. Where it's been implemented (`DashItem`, `PluginDashItem`, and `SettingsCard`) `label` is always passed, which makes sense here given how its used. Have updated the prop to be required. Thank you!

Comment (jeyip) [reply projects/plugins/jetpack/_inc/client/components/section-header/index.tsx]: Thank you! [ba7f535](https://github.com/Automattic/jetpack/pull/26644/commits/ba7f535f684731b4bf5b491b4274301907aa37d9) looks good to me 👍

**Comment by kirahsapong** [comment]
> For the typed section header file, I'm seeing a "can't find module" error. Did you / do you happen to see the same thing?

No, everything looks good on my end! Have you tried building on your end?

**Comment by jeyip** [comment]
> No, everything looks good on my end! Have you tried building on your end?

I'll give it a shot. Either way, it must be something on my end. I vaguely remember seeing something similar a while ago on one of my PRs where typescript annotations weren't being updated in build files.

**Review by jeyip — DISMISSED** [review]

**Review by jeyip — DISMISSED** [review]

**Review by sharpirate — APPROVED** [review]
✅ This looks good on my end 

image

**Review by bindlegirl — APPROVED** [review]

**Pull request merged**